### ISSUE TITLE
on_load function execute twice.

### ISSUE THREAD

**Issue by latestk-develop** [reflex-dev/reflex#1642]
**Describe the bug**
It is normally called only once right after compilation. But after that, it is called twice.

**To Reproduce**
```from rxconfig import config

import reflex as rx

import time
class State(rx.State):
    market_id: str = ""
    
    def load_market(self):
        print(f'load market {time.time()}')
        self.market_id = self.get_query_params().get('mid')

@rx.page('/market/[mid]', on_load=State.load_market)
def market():
    return rx.vstack(
        rx.heading(State.market_id),
        rx.button(
            "return",
            on_click=rx.redirect("/"),
        )
    )
def index() -> rx.Component:
    return rx.fragment(
        rx.vstack(
            rx.button(
                "Load Market 1",
                on_click=rx.redirect("/market/1"),
            )
        ),
    )


# Add state and page to the app.
app = rx.App()
app.add_page(index)
app.compile()
```

Compiling:  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 2/1 0:00:00
load market 1692600836.5542612 -> first ( click Load Market 1 button)
then  click return button in /market/1 and click Load Market 1 button
load market 1692600840.4809003 -> second
load market 1692600840.4879425 -> second

**Expected behavior**

**Screenshots**

**Specifics (please complete the following information):**
 - Python Version: 3.11.4
 - Reflex Version: 0.2.4
 - OS: Windows
 - Browser (Optional):
 
**Additional context**

**Comment by Lendemor** [comment]
Does it happens in dev mode? prod mode? both?

**Comment by latestk-develop** [comment]
Dev mode. Command “reflex run”

**Comment by wassafshahzad** [comment]
I can look into this issue.

**Comment by masenf** [comment]
This will be fixed in 0.2.5!

**Comment by danik292** [comment]
@masenf @latestk-develop can we test it in a new version and meybe close it?
